### PR TITLE
Optimize user queries

### PR DESCRIPTION
**Pull request by tung2744**
ref #3233 

If you are interested in the actual time consumed in each interaction nodes, you can try this branch which added logs to show the duration of each steps:
https://github.com/tung2744/authgear-server/tree/3233-logs

### CODE REVIEW

**Comment by tung2744** [comment]
Sorry, I found it break latte flows. Mark this pr as WIP first.

**Comment by tung2744** [comment]
Added a commit to fix a bug in authenticator service.

File: pkg/lib/feature/stdattrs/service_noevent.go [hunk 1]
```diff
@@ -193,63 +194,102 @@ func (s *ServiceNoEvent) UpdateStandardAttributes(role accesscontrol.Role, userI
...
-		if err != nil {
-			return nil, err
-		}
+	if len(userIDs) != len(updatedAts) || len(userIDs) != len(attrsList) {
+		panic("stdattrs: expeceted same length of arguments")
+	}
 
-		// Copy
-		out[key] = value
+	allClaims, err := s.ClaimStore.ListByUserIDsAndClaimNames(
```
Comment (louischan-oursky) [inline pkg/lib/feature/stdattrs/service_noevent.go:209]: This slice contains claims of different users. If there are 100 hundreds, and each user has an email address and a phone number, this slice is of length 200. And the the following loop has at most 100 * 200 iterations.

Maybe we can group `allClaims` into `claimsByUserID map[string][]*Claim` first? Then we need 200 iterations to group and 100 iterations to build the results.

File: pkg/lib/authn/authenticator/service/service.go [hunk 2]
```diff
@@ -235,11 +270,33 @@ func (s *Service) List(userID string, filters ...authenticator.Filter) ([]*authe
...
 		}
 		if keep {
-			filtered = append(filtered, a)
+			filteredInfos = append(filteredInfos, a)
 		}
 	}
 
-	return filtered, nil
+	infosByUserID := map[string][]*authenticator.Info{}
+	for _, info := range infos {
```
[the remark this answers is not in the document]
Comment (tung2744) [reply pkg/lib/authn/authenticator/service/service.go]: Fixed in [1bc85d7](https://github.com/authgear/authgear-server/pull/3400/commits/1bc85d7880acf85979f3219ca6222b6fa7e1a19d)

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

**Pull request merged**